src/main/java/org/example/dao/PointDAOImpl.java

```diff
@@ -0,0 +1,95 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class PointDAOImpl implements PointDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public PointDAOImpl() {
+    }
+
+    @Override
+    public void insertPoint(Point point) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try (PreparedStatement statement = connection.prepareStatement("INSERT INTO points (id, xCoordinate, yCoordinate) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS)) {
+            statement.setInt(1, point.getId());
+            statement.setDouble(2, point.getXCoordinate());
+            statement.setDouble(3, point.getYCoordinate());
+            statement.executeUpdate();
+
+            try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                while (generatedKeys.next()) {
+                    int id = generatedKeys.getInt(1);
+                    point.setId(id);
+                    System.out.println("Point inserted successfully. Generated ID: " + id);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+    @Override
+    public void deletePoint(int id) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try (PreparedStatement statement = connection.prepareStatement("DELETE FROM points WHERE id = ?")) {
+            statement.setInt(1, id);
+            statement.executeUpdate();
+            System.out.println("Point with ID " + id + " deleted successfully.");
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+
+    @Override
+    public Point getPoint(int id) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        Point point = null;
+        try (PreparedStatement statement = connection.prepareStatement("SELECT  xCoordinate, yCoordinate FROM points WHERE id = ?")) {
```

Comment:
This one should be
SELECT x_coordinate AS xCoordinate, y_coordinate AS yCoordinate FROM points WHERE id = ?